docker/entrypoint.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh -e
+
+if [ "$(id -u)" != 0 ]; then
+    echo "Already running as non-root, can't change user."
+    exec "$@"
+fi
+# Create a non-root user with UID/GID matching the host user to ensure that
+# files written to the volumes are not owned by root.
+: "${cheribuild_uid:=1234}"
+: "${cheribuild_gid:=1234}"
+: "${cheribuild_user:=cheri}"
+addgroup --quiet --gid ${cheribuild_gid} "${cheribuild_user}"
+yes | adduser --quiet --uid ${cheribuild_uid} --disabled-password --ingroup "${cheribuild_user}" "${cheribuild_user}" > /dev/null
```

Comment:
What's the yes for?

The --disabled-password is fine, unless we want to put the user in the sudo group, at which point they'll need a password (blank generally easiest in this kind of situation...). I don't think you can set that with adduser, so you'd either need to set their password as an extra step or switch to using the lower-level useradd (and I guess groupadd for symmetry) commands.

Reply:
For the `Is the information correct? [Y/n]` prompt.

However it seems like setting stdin to /dev/null also works (but prints warnings):
```
docker run -it ubuntu:latest
root@3094ac014bb2:/# addgroup --quiet --gid 501 alex && adduser --uid 501 --disabled-password --ingroup alex alex < /dev/null
Adding user `alex' ...
Adding new user `alex' (501) with group `alex' ...
Creating home directory `/home/alex' ...
Copying files from `/etc/skel' ...
Changing the user information for alex
Enter the new value, or press ENTER for the default
	Full Name []: 	Room Number []: 	Work Phone []: 	Home Phone []: Other []: Use of uninitialized value $answer in chop at /usr/sbin/adduser line 621.
Use of uninitialized value $answer in pattern match (m//) at /usr/sbin/adduser line 622.
Is the information correct? [Y/n]
```

Reply:
Changed to use useradd